Add Elastic Agent 9.5.0 release notes (#15826)
elastic/elastic-agent: docs/release-notes/_snippets/9.5.0/index.md

## 9.5.0 [elastic-agent-release-notes-9.5.0]

### Features and enhancements [elastic-agent-9.5.0-features-enhancements]

> * Enable AWS CloudWatch metrics ingestion via the awscloudwatchreceiver in the EDOT Collector. [#14870](https://github.com/elastic/elastic-agent/pull/14870) 

Reviewer: ```suggestion
* Enable AWS CloudWatch metrics ingestion through `awscloudwatchreceiver` in the EDOT Collector. [#14870](https://github.com/elastic/elastic-agent/pull/14870) 
```

> * Add Azure Event Hub receiver to EDOT Collector. [#13418](https://github.com/elastic/elastic-agent/pull/13418) 

Reviewer: ```suggestion
* Enable AWS CloudWatch metrics ingestion through `awscloudwatchreceiver` in the EDOT Collector. [#14870](https://github.com/elastic/elastic-agent/pull/14870) 
* Add Azure Event Hub receiver to EDOT Collector. [#13418](https://github.com/elastic/elastic-agent/pull/13418) 
```

> * Make otel runtime the default for filebeat inputs. [#13991](https://github.com/elastic/elastic-agent/pull/13991) 

Reviewer: ```suggestion
* Set OTel as the default runtime for Filebeat inputs. [#13991](https://github.com/elastic/elastic-agent/pull/13991) 
```

> * Add support for auditbeat, heartbeat, osquerybeat, and packetbeat to run under otel mode. [#14029](https://github.com/elastic/elastic-agent/pull/14029) 

Reviewer: ```suggestion
* Add support for Auditbeat, Heartbeat, Osquerybeat, and Packetbeat to run in `otel` mode. [#14029](https://github.com/elastic/elastic-agent/pull/14029) 
```

* Add Akamai SIEM receiver to EDOT Collector. [#14263](https://github.com/elastic/elastic-agent/pull/14263) 
* Add AWS CloudWatch receiver to EDOT Collector. [#14774](https://github.com/elastic/elastic-agent/pull/14774) 

> * Add credentials provider extension for use with awscloudwatchreceiver in edot. [#14933](https://github.com/elastic/elastic-agent/pull/14933) 

Reviewer: ```suggestion
* Add credentials provider extension for use with `awscloudwatchreceiver` in EDOT Collector. [#14933](https://github.com/elastic/elastic-agent/pull/14933) 
```

> * Update awscloudwatchreceiver to support cross-account monitoring for metrics. [#15085](https://github.com/elastic/elastic-agent/pull/15085) 

Reviewer: ```suggestion
* Update `awscloudwatchreceiver` to support cross-account monitoring for metrics. [#15085](https://github.com/elastic/elastic-agent/pull/15085) 
```

> * Update edot awscloudwatchreceiver to support `RecentlyActive` for metrics discovery. [#15121](https://github.com/elastic/elastic-agent/pull/15121) 

Reviewer: ```suggestion
* Update `awscloudwatchreceiver` in EDOT Collector to support `RecentlyActive` for metrics discovery. [#15121](https://github.com/elastic/elastic-agent/pull/15121) 
```

* Run audit inputs as collector receivers by default, reducing steady state memory usage. [#15188](https://github.com/elastic/elastic-agent/pull/15188) 

> * Run osquery inputs as collector receivers by default, reducing steady state memory usage. [#15459](https://github.com/elastic/elastic-agent/pull/15459) 

Reviewer: ```suggestion
* Run Osquery inputs as collector receivers by default, reducing steady state memory usage. [#15459](https://github.com/elastic/elastic-agent/pull/15459) 
```

* Write OTel collector logs to its own log file. [#15491](https://github.com/elastic/elastic-agent/pull/15491) 

  When running under the otel runtime, the OTel collector now writes to its
  own log file instead of the main agent log file. Its event logs go to a
  separate collector event log file rather than the agent&#39;s event log.

>   

Reviewer: ```suggestion
* Write OTel Collector logs to its own log file. [#15491](https://github.com/elastic/elastic-agent/pull/15491)
```

* Run each stream in its own receiver (OTEL mode). [#13000](https://github.com/elastic/elastic-agent/pull/13000) 

  When an input is running in OTEL mode each stream is defined as its own receiver.

>   

Reviewer: ```suggestion
* Run each stream in its own receiver (`otel` mode). [#13000](https://github.com/elastic/elastic-agent/pull/13000)
```

* Update OTel Collector components to v0.149.0. [#13599](https://github.com/elastic/elastic-agent/pull/13599) 
* Update OTel Collector components to v0.150.0/v1.56.0. [#13696](https://github.com/elastic/elastic-agent/pull/13696) 

> * Enrich APM data with k8s attributes at the edot daemon collector. [#13746](https://github.com/elastic/elastic-agent/pull/13746) 

Reviewer: ```suggestion
* Enrich APM data with Kubernetes attributes in the EDOT daemon collector. [#13746](https://github.com/elastic/elastic-agent/pull/13746) 
```

> * Enrich APM data with k8s attributes at the edot daemon collector in non managed kube-stack. [#13901](https://github.com/elastic/elastic-agent/pull/13901) 

Reviewer: ```suggestion
* Enrich APM data with Kubernetes attributes in the EDOT daemon collector in non managed kube-stack. [#13901](https://github.com/elastic/elastic-agent/pull/13901) 
```

> * Include fleet-server in windows/arm64 packages. [#14120](https://github.com/elastic/elastic-agent/pull/14120) 

Reviewer: ```suggestion
* Include `fleet-server` in `windows/arm64` packages. [#14120](https://github.com/elastic/elastic-agent/pull/14120) 
```

* Update OTel Collector components to v0.152.0. [#14198](https://github.com/elastic/elastic-agent/pull/14198) 

> * Expose buffer_size option on StateWatch RPC to skip intermediate states. [#14257](https://github.com/elastic/elastic-agent/pull/14257) 

Reviewer: ```suggestion
* Expose `buffer_size` option on StateWatch RPC to skip intermediate states. [#14257](https://github.com/elastic/elastic-agent/pull/14257) 
```

* Periodic cleanup now removes leftover directories from failed upgrades. [#14378](https://github.com/elastic/elastic-agent/pull/14378) 

  The periodic cleanup previously only removed directories tracked by the TTL
  registry. Directories left behind by failed upgrades — those that never got
  a .ttl marker written — were invisible to it and only removed when the next
  upgrade ran. The cleanup now scans all elastic-agent-* directories and
  removes any that are not the current home, referenced by an active upgrade
  marker, or covered by an unexpired TTL entry.
  
  Additionally, the upgrade marker is now written before unpacking begins so
  that the target directory is protected from a concurrent cleanup run
  throughout the entire upgrade process.

>   

Reviewer: ```suggestion
* Periodic cleanup now removes leftover directories from failed upgrades. [#14378](https://github.com/elastic/elastic-agent/pull/14378)
```

* Write EDOT internal telemetry to a dedicated diagnostics file. [#14506](https://github.com/elastic/elastic-agent/pull/14506) [#14446](https://github.com/elastic/elastic-agent/issues/14446)
* Read manual rollback targets from on-disk TTL files instead of the upgrade marker. [#14543](https://github.com/elastic/elastic-agent/pull/14543) 
* Update OTel Collector components to v0.155.0. [#15333](https://github.com/elastic/elastic-agent/pull/15333) 

> * Use native GO for linux FIPS builds. [#14953](https://github.com/elastic/elastic-agent/pull/14953) 

Reviewer: ```suggestion
* Use native Go for Linux FIPS builds. [#14953](https://github.com/elastic/elastic-agent/pull/14953) 
```

> * Bump kube-stack to 0.16.0 and fix ELASTIC_AGENT_OTEL quoting for otel mode. [#15013](https://github.com/elastic/elastic-agent/pull/15013) 

Reviewer: ```suggestion
* Bump kube-stack to v0.16.0 and fix `ELASTIC_AGENT_OTEL` quoting for `otel` mode. [#15013](https://github.com/elastic/elastic-agent/pull/15013) 
```

* Set OTel as the default runtime for Packetbeat. [#15151](https://github.com/elastic/elastic-agent/pull/15151) [#14554](https://github.com/elastic/elastic-agent/issues/14554)
* Set OTel as the default runtime for Heartbeat. [#15325](https://github.com/elastic/elastic-agent/pull/15325) 
* Add environment variables to collector diagnostics.  

### Fixes [elastic-agent-9.5.0-fixes]

* Redact sensitive HTTP header values embedded in Fleet env vars in diagnostics archives. [#15284](https://github.com/elastic/elastic-agent/pull/15284) 

  Diagnostics archives could expose authentication tokens when Fleet header environment
  variables (such as FLEET_HEADER, FLEET_HEADERS, FLEET_KIBANA_HEADER, and
  FLEET_KIBANA_HEADERS) contained sensitive HTTP header values. The diagnostics redaction
  logic now expands and redacts individual header values within these variables.

>   

Reviewer: ```suggestion
* Redact sensitive HTTP header values embedded in Fleet environment variables in diagnostics archives. [#15284](https://github.com/elastic/elastic-agent/pull/15284)
```

* Ensure current runtime stops before new runtime starts. [#13584](https://github.com/elastic/elastic-agent/pull/13584) 

  Ensures that the current runtime stops before the new runtime starts, otherwise registry
  state will not be consistent before the start of the new runtime.

>   

Reviewer: ```suggestion
* Ensure the current runtime stops before the new runtime starts. [#13584](https://github.com/elastic/elastic-agent/pull/13584)
```

* Fix uncleared config manager errors permanently failing aggregate state. [#14042](https://github.com/elastic/elastic-agent/pull/14042) 
* Tolerate corrupt `.ttl` rollback markers during upgrade and rollback. [#14143](https://github.com/elastic/elastic-agent/pull/14143) 

  A corrupt `.ttl` rollback marker file in the live agent install previously
  caused the TTL registry to abort with a parse error, which blocked every
  subsequent upgrade and rollback attempt. The registry now tolerates
  malformed existing markers: corrupt entries in the desired state are
  overwritten with valid YAML, and entries absent from the desired state are
  swept from disk, allowing upgrade and rollback flows to self-heal.

>   

Reviewer: ```suggestion
* Tolerate corrupt `.ttl` rollback markers during upgrade and rollback. [#14143](https://github.com/elastic/elastic-agent/pull/14143)
```

* Fix duplicate entries, empty unit dirs, and EDOT error handling in OTel diagnostics. [#15108](https://github.com/elastic/elastic-agent/pull/15108) 

  The OTel diagnostics ZIP no longer contains duplicate entries and no longer creates empty unit subdirectories. Components with no EDOT diagnostics no longer produce a spurious error in the archive. Also, an unexpected EDOT error used to abort the whole component-diagnostics request; now it is recorded per component so the diagnostics archive is still produced.

>   

Reviewer: ```suggestion
* Fix duplicate entries, empty unit dirs, and EDOT error handling in OTel diagnostics. [#15108](https://github.com/elastic/elastic-agent/pull/15108)
```

> * Fix minimum batch size for supervised OTel collector ES and Kafka exporters. [#15122](https://github.com/elastic/elastic-agent/pull/15122) [#15118](https://github.com/elastic/elastic-agent/issues/15118)

Reviewer: ```suggestion
* Fix the minimum batch size for supervised OTel Collector Elasticsearch and Kafka exporters. [#15122](https://github.com/elastic/elastic-agent/pull/15122) [#15118](https://github.com/elastic/elastic-agent/issues/15118)
```

> * Propagate the `agent.features.fqdn.enabled` feature flag to beat receivers. [#15191](https://github.com/elastic/elastic-agent/pull/15191) [#15165](https://github.com/elastic/elastic-agent/issues/15165)

Reviewer: ```suggestion
* Propagate the `agent.features.fqdn.enabled` feature flag to Beat receivers. [#15191](https://github.com/elastic/elastic-agent/pull/15191) [#15165](https://github.com/elastic/elastic-agent/issues/15165)
```

* Preserve locally-configured monitoring.http.host across Fleet policy check-ins. [#15291](https://github.com/elastic/elastic-agent/pull/15291) 

  A regression introduced in 9.4.2 caused the agent&#39;s monitoring HTTP listener to rebind to the
  default host (localhost) on every Fleet policy check-in, discarding any host configured locally
  via agent.monitoring.http.host (e.g. 0.0.0.0). The policy-change handler now only applies
  monitoring.http/pprof settings from the policy when they are explicitly present, leaving the
  locally-configured values untouched otherwise.

>   

Reviewer: ```suggestion
* Preserve locally-configured `monitoring.http.host` across Fleet policy check-ins. [#15291](https://github.com/elastic/elastic-agent/pull/15291)
```

* Fix translation of `queue.mem.flush.timeout` to OTel `flush_timeout` for unitless values. [#15404](https://github.com/elastic/elastic-agent/pull/15404) 

  When translating an output&#39;s Beats configuration to an OpenTelemetry Collector
  configuration, a unitless queue.mem.flush.timeout value (e.g. `5`, which Beats
  interprets as 5 seconds) was passed through verbatim as a string to the exporterhelper
  sending_queue.batch.flush_timeout option. Because that option is a time.Duration,
  the collector failed to start with &#34;&#39;flush_timeout&#39; time: missing unit in duration&#34;.
  Unitless values are now suffixed with the seconds unit.

>   

Reviewer: ```suggestion
* Fix translation of `queue.mem.flush.timeout` to OTel `flush_timeout` for unitless values. [#15404](https://github.com/elastic/elastic-agent/pull/15404)
```

> * Fixed FIPS mode accepting non-compliant TLS certificates. [#14986](https://github.com/elastic/elastic-agent/pull/14986) 

Reviewer: ```suggestion
* Fix an issue where FIPS mode accepted non-compliant TLS certificates. [#14986](https://github.com/elastic/elastic-agent/pull/14986) 
```

> * Fix spurious &#34;failed to unmarshal checkin actions&#34; error on idle Fleet check-ins. [#15398](https://github.com/elastic/elastic-agent/pull/15398) [#15397](https://github.com/elastic/elastic-agent/issues/15397)

Reviewer: ```suggestion
* Fix spurious "failed to unmarshal checkin actions" error on idle Fleet check-ins. [#15398](https://github.com/elastic/elastic-agent/pull/15398) [#15397](https://github.com/elastic/elastic-agent/issues/15397)
```

> * Override fleet.ssl.certificate_authorities from env vars in container mode. [#15427](https://github.com/elastic/elastic-agent/pull/15427) 

Reviewer: ```suggestion
* Override `fleet.ssl.certificate_authorities` from environment variables in container mode. [#15427](https://github.com/elastic/elastic-agent/pull/15427) 
```

## 9.5.0 [elastic-agent-release-notes-9.5.0]

### Features and enhancements [elastic-agent-9.5.0-features-enhancements]

* Enable AWS CloudWatch metrics ingestion through `awscloudwatchreceiver` in the EDOT Collector. [#14870](https://github.com/elastic/elastic-agent/pull/14870)
* Add Azure Event Hub receiver to EDOT Collector. [#13418](https://github.com/elastic/elastic-agent/pull/13418)

> * Set OTel as the default runtime for Filebeat inputs. [#13991](https://github.com/elastic/elastic-agent/pull/13991)

Reviewer 2: ```suggestion
* Run Filebeat based inputs as collector receivers by default, reducing steady state memory usage [#13991](https://github.com/elastic/elastic-agent/pull/13991)
```

> * Add support for Auditbeat, Heartbeat, Osquerybeat, and Packetbeat to run in `otel` mode. [#14029](https://github.com/elastic/elastic-agent/pull/14029)

Reviewer 2: ```suggestion
* Add support for Auditbeat, Heartbeat, Osquerybeat, and Packetbeat based inputs to run in `otel` mode. [#14029](https://github.com/elastic/elastic-agent/pull/14029)
* Run synthetics inputs as collector receivers by default, reducing steady state memory usage. [#15325](https://github.com/elastic/elastic-agent/pull/15325)
* Run packet inputs as collector receivers by default, reducing steady state memory usage. [#15151](https://github.com/elastic/elastic-agent/pull/15151) [#14554](https://github.com/elastic/elastic-agent/issues/14554)
* Run audit inputs as collector receivers by default, reducing steady state memory usage. [#15188](https://github.com/elastic/elastic-agent/pull/15188)
* Run Osquery inputs as collector receivers by default, reducing steady state memory usage. [#15459](https://github.com/elastic/elastic-agent/pull/15459)
```

* Add Akamai SIEM receiver to EDOT Collector. [#14263](https://github.com/elastic/elastic-agent/pull/14263)
* Add AWS CloudWatch receiver to EDOT Collector. [#14774](https://github.com/elastic/elastic-agent/pull/14774)
* Add credentials provider extension for use with `awscloudwatchreceiver` in EDOT Collector. [#14933](https://github.com/elastic/elastic-agent/pull/14933)
* Update `awscloudwatchreceiver` to support cross-account monitoring for metrics. [#15085](https://github.com/elastic/elastic-agent/pull/15085)
* Update `awscloudwatchreceiver` in EDOT Collector to support `RecentlyActive` for metrics discovery. [#15121](https://github.com/elastic/elastic-agent/pull/15121)
* Run audit inputs as collector receivers by default, reducing steady state memory usage. [#15188](https://github.com/elastic/elastic-agent/pull/15188)

> * Run Osquery inputs as collector receivers by default, reducing steady state memory usage. [#15459](https://github.com/elastic/elastic-agent/pull/15459)

Reviewer 2: ```suggestion
```

* Include `fleet-server` in `windows/arm64` packages. [#14120](https://github.com/elastic/elastic-agent/pull/14120)
* Expose `buffer_size` option on StateWatch RPC to skip intermediate states. [#14257](https://github.com/elastic/elastic-agent/pull/14257)
* Periodic cleanup now removes leftover directories from failed upgrades. [#14378](https://github.com/elastic/elastic-agent/pull/14378)
* Write Elastic Agent self-monitoring metrics to a dedicated diagnostics file in the OTLP metrics format. [#14506](https://github.com/elastic/elastic-agent/pull/14506) [#14446](https://github.com/elastic/elastic-agent/issues/14446)
* Read manual rollback targets from on-disk TTL files instead of the upgrade marker. [#14543](https://github.com/elastic/elastic-agent/pull/14543)
* Update OTel Collector components to v0.155.0. [#15333](https://github.com/elastic/elastic-agent/pull/15333)
* Use native Go for Linux FIPS builds. [#14953](https://github.com/elastic/elastic-agent/pull/14953)
* Bump kube-stack to v0.16.0 and fix `ELASTIC_AGENT_OTEL` quoting for `otel` mode. [#15013](https://github.com/elastic/elastic-agent/pull/15013)

> * Set OTel as the default runtime for Packetbeat. [#15151](https://github.com/elastic/elastic-agent/pull/15151) [#14554](https://github.com/elastic/elastic-agent/issues/14554)

Reviewer 2: ```suggestion
```

> * Set OTel as the default runtime for Heartbeat. [#15325](https://github.com/elastic/elastic-agent/pull/15325)

Reviewer 2: ```suggestion
```

## 9.5.0 [elastic-agent-release-notes-9.5.0]

### Features and enhancements [elastic-agent-9.5.0-features-enhancements]

* Enable AWS CloudWatch metrics ingestion through `awscloudwatchreceiver` in the EDOT Collector. [#14870](https://github.com/elastic/elastic-agent/pull/14870) 
* Add Azure Event Hub receiver to EDOT Collector. [#13418](https://github.com/elastic/elastic-agent/pull/13418) 

> * Set OTel as the default runtime for Filebeat inputs. [#13991](https://github.com/elastic/elastic-agent/pull/13991) 

Reviewer 2: ```suggestion
* Set OTel as the default runtime for Filebeat inputs, reducing steady state memory usage. [#13991](https://github.com/elastic/elastic-agent/pull/13991) 
```

> * Add support for Auditbeat, Heartbeat, Osquerybeat, and Packetbeat to run in `otel` mode. [#14029](https://github.com/elastic/elastic-agent/pull/14029) 

Reviewer 2: We should group the notes indicating these all run in the otel runtime by default here too.

Author: @vishaangelova any feedback on how to best handle formatting to group as @cmacknz suggests?

* Add Akamai SIEM receiver to EDOT Collector. [#14263](https://github.com/elastic/elastic-agent/pull/14263) 
* Add AWS CloudWatch receiver to EDOT Collector. [#14774](https://github.com/elastic/elastic-agent/pull/14774) 
* Add credentials provider extension for use with `awscloudwatchreceiver` in EDOT Collector. [#14933](https://github.com/elastic/elastic-agent/pull/14933) 
* Update `awscloudwatchreceiver` to support cross-account monitoring for metrics. [#15085](https://github.com/elastic/elastic-agent/pull/15085) 
* Update `awscloudwatchreceiver` in EDOT Collector to support `RecentlyActive` for metrics discovery. [#15121](https://github.com/elastic/elastic-agent/pull/15121) 
* Run audit inputs as collector receivers by default, reducing steady state memory usage. [#15188](https://github.com/elastic/elastic-agent/pull/15188) 
* Run Osquery inputs as collector receivers by default, reducing steady state memory usage. [#15459](https://github.com/elastic/elastic-agent/pull/15459) 
* Write OTel Collector logs to its own log file. [#15491](https://github.com/elastic/elastic-agent/pull/15491)

> * Run each stream in its own receiver (`otel` mode). [#13000](https://github.com/elastic/elastic-agent/pull/13000)

Reviewer 2: ```suggestion
* Run each per Beat input stream in it's own OpenTelemetry collector receiver (`otel` mode), allowing hot reloading of receiver configurations without collector restarts [#13000](https://github.com/elastic/elastic-agent/pull/13000)
```

> * Update OTel Collector components to v0.149.0. [#13599](https://github.com/elastic/elastic-agent/pull/13599) 

Reviewer 2: ```suggestion
```

We only need the last one, not any of the intermediate updates.

> * Update OTel Collector components to v0.150.0/v1.56.0. [#13696](https://github.com/elastic/elastic-agent/pull/13696) 

Reviewer 2: ```suggestion
```

* Enrich APM data with Kubernetes attributes in the EDOT daemon collector. [#13746](https://github.com/elastic/elastic-agent/pull/13746) 
* Enrich APM data with Kubernetes attributes in the EDOT daemon collector in non managed kube-stack. [#13901](https://github.com/elastic/elastic-agent/pull/13901) 
* Include `fleet-server` in `windows/arm64` packages. [#14120](https://github.com/elastic/elastic-agent/pull/14120) 

> * Update OTel Collector components to v0.152.0. [#14198](https://github.com/elastic/elastic-agent/pull/14198) 

Reviewer 2: ```suggestion
```

* Expose `buffer_size` option on StateWatch RPC to skip intermediate states. [#14257](https://github.com/elastic/elastic-agent/pull/14257) 
* Periodic cleanup now removes leftover directories from failed upgrades. [#14378](https://github.com/elastic/elastic-agent/pull/14378)

> * Write EDOT internal telemetry to a dedicated diagnostics file. [#14506](https://github.com/elastic/elastic-agent/pull/14506) [#14446](https://github.com/elastic/elastic-agent/issues/14446)

Reviewer 2: ```suggestion
* Write Elastic Agent self-monitoring metrics to a dedicated diagnostics file in the OTLP metrics format. [#14506](https://github.com/elastic/elastic-agent/pull/14506) [#14446](https://github.com/elastic/elastic-agent/issues/14446)
```

* Read manual rollback targets from on-disk TTL files instead of the upgrade marker. [#14543](https://github.com/elastic/elastic-agent/pull/14543) 
* Update OTel Collector components to v0.155.0. [#15333](https://github.com/elastic/elastic-agent/pull/15333) 

> * Use native Go for Linux FIPS builds. [#14953](https://github.com/elastic/elastic-agent/pull/14953) 

Reviewer 2: ```suggestion
* Use the FIPS 140-3 certified Go crypto module for Linux FIPS builds, removing runtime dependency on OpenSSL. [#14953](https://github.com/elastic/elastic-agent/pull/14953) 
```

* Bump kube-stack to v0.16.0 and fix `ELASTIC_AGENT_OTEL` quoting for `otel` mode. [#15013](https://github.com/elastic/elastic-agent/pull/15013) 
* Set OTel as the default runtime for Packetbeat. [#15151](https://github.com/elastic/elastic-agent/pull/15151) [#14554](https://github.com/elastic/elastic-agent/issues/14554)
* Set OTel as the default runtime for Heartbeat. [#15325](https://github.com/elastic/elastic-agent/pull/15325) 

> * Add environment variables to collector diagnostics.  

Reviewer: This one doesn’t list a PR.

Author: Fixed in [9cd6254](https://github.com/elastic/elastic-agent/pull/15826/commits/9cd62543a9558cc82dd3faa0c28e6ac3afb01bab)
